lib/imagery/sigfile.c

```diff
@@ -17,56 +17,42 @@
 /*!
    \brief Create signature file
 
-   \param group group name
-   \param subgroup subgroup name in given group
    \param name signature filename
 
-   \return pointer to FILE*
+   \return pointer to FILE
    \return NULL on error
 */
-FILE *I_fopen_signature_file_new(const char *group,
-				 const char *subgroup, const char *name)
+FILE *I_fopen_signature_file_new(const char *name)
 {
-    char element[GPATH_MAX];
-    char group_name[GNAME_MAX], group_mapset[GMAPSET_MAX];
     FILE *fd;
 
-    if (!G_name_is_fully_qualified(group, group_name, group_mapset)) {
-	strcpy(group_name, group);
-    }
+    /* create sig directory */
+    G__make_mapset_element_misc("signatures", "sig");
```

Comment:
Wouldn't a plain G_make_mapset_dir_object("signatures", "sig"); work?

Reply:
No, the second parameter is name such as bridges or elevation, so the creation race behavior for that is different than for your element/subelement case. Two processes can try to create element/subelement at the same time and all what matters is that the directory exists when the function returns. That's what G_make_mapset_object_group does for element. Since there should be only one process creating directory called _name_ (bridges, elevation), G_make_mapset_dir_object fails when the directory failed to create even if the reason was that it already exist.

This is the difference between using these function as "create some nested dirs" versus "create a specific object in a mapset".  In the first case, any function would do and you could pass any `aaa/bbb/ccc/.../zzz` as either element or name. In the second case, you need a specialized function for each case to properly express what you are doing which makes it possible for the lower level functions to do the most appropriate thing in the given context (and it also makes it possible to change that behavior when it turns out to be incorrect).

This is mostly lessons learned in #1681.

Reply:
I reworked the element creation to mimic vector lib approach with an internal function. Now it should be enough for a temporary workaround.